src/main/java/pl/asie/ynot/oc/OCChannelSettings.java

```diff
@@ -52,10 +61,15 @@ public OCChannelSettings() {
     }
 
     private Node getNode(World world, BlockPos pos, EnumFacing side) {
-        if (world.getTileEntity(pos) instanceof SidedEnvironment) {
-            return ((SidedEnvironment) world.getTileEntity(pos)).sidedNode(side.getOpposite());
-        } else if(world.getTileEntity(pos) instanceof Environment) {
-            return ((Environment) world.getTileEntity(pos)).node();
+        TileEntity tile = world.getTileEntity(pos);
+        if(tile == null) {
+            return null;
+        }
+
+        if (tile.hasCapability(SIDED_ENVIRONMENT_CAPABILITY, side)) {
+            return tile.getCapability(SIDED_ENVIRONMENT_CAPABILITY, side).sidedNode(side.getOpposite());
```

Comment:
Are you sure this is correct? You know OC's APIs better than I do - did you verify?

Reply:
At the very least, it's working for server racks + computers + some built-in components. I'll double-check with Vex to see if this is right.

Reply:
Vex says: `10:49:31 <+Vexatos> all TEs implementing those have the cap as well`